frame/support/procedural/src/pallet/parse/trait_.rs

```diff
@@ -148,8 +149,43 @@ impl TraitDef {
 			}
 		}
 
-		// TODO TODO: check for frame_system bound ?
+		let has_frame_system_supertrait = item.supertraits.iter().any(|s| {
+			syn::parse2::<FrameSystemTraitParse>(s.to_token_stream()).is_ok()
+		});
+
+		if !has_frame_system_supertrait {
```

Comment:
So this macro cannot be used on `frame_system` itself?

Reply:
ah indeed.
I started to requires an explicit `frame_system::Trait` because if user forgot to bound it then the error message was difficult to read.
I will add an attribute: `#[pallet::has_frame_system_trait]` so that this is not always required